testing/functional_tests/test_automate_training.py

```diff
@@ -34,6 +35,38 @@ def test_automate_training(download_functional_test_files, script_runner):
     check_sha256(str(file_config))
 
 
+def test_automate_training_run_test_debug(download_functional_test_files):
+    """A unit test similar to test_automate_training_run_test but allow step through (instead of using script caller/
+    subprocess mode which cannot be stepped. Other than that, nothing else really changed and is exactly the same.
+    Very useful for debugging this high level function to spot problems
+
+    Fixture Required:
+        download_functional_test_files:
+    """
+    file_config = os.path.join(__data_testing_dir__, 'automate_training_config.json')
+    file_config_hyper = os.path.join(__data_testing_dir__,
+                                     'automate_training_hyperparameter_opt.json')
+    __output_dir__ = os.path.join(__tmp_dir__, 'results')
+
+    from ivadomed.scripts.automate_training import automate_training
+
+    automate_training(file_config=file_config,
+                      file_config_hyper=file_config_hyper,
+                      path_data=__data_testing_dir__,
+                      run_test=True,
+                      output_dir=__output_dir__,
+                      fixed_split=False,
+                      all_combin=True,
+                      n_iterations=1,
+                      all_logs=True,
+                      multi_params=True,
+                      )
+
+    assert os.path.exists(os.path.join(__output_dir__, 'detailed_results.csv'))
+    assert os.path.exists(os.path.join(__output_dir__, 'temporary_results.csv'))
+    assert os.path.exists(os.path.join(__output_dir__, 'average_eval.csv'))
```

Comment:
Maybe `Path` and `Path.exists()`

Reply:
Great point. Updated. Much easier to read.